### PR TITLE
[6052] Dashboard responsive updates

### PR DESCRIPTION
**Pull request by jvega190**
https://github.com/craftercms/craftercms/issues/6052

### CODE REVIEW

File: ui/app/src/components/DashletCard/DashletCard.tsx [hunk 1]
```diff
@@ -40,10 +48,13 @@ export type DashletCardProps = PropsWithChildren<
       footer: BoxProps['sx'];
     }>;
     cardContentProps?: CardContentProps;
+    maximizable?: boolean;
+    collapsible?: boolean;
```
Comment (jvega190) [inline ui/app/src/components/DashletCard/DashletCard.tsx]: Collapsible is needed for it not to be shown in the activityDashlet (in desktopMode). Defaults to true

File: ui/app/src/components/DashletCard/DashletCard.tsx [hunk 2]
```diff
@@ -68,62 +94,118 @@ export function DashletCard(props: DashletCardProps) {
...
+                </>
+              }
+              sx={{ pr: 1, width: '100%', ...sxs?.header }}
+            />
+          </AccordionSummary>
+        )}
+        {/* endregion */}
+        {!isMaximized && <AccordionDetails sx={{ p: 0 }}>{renderCardContent()}</AccordionDetails>}
+      </Accordion>
+      {isMaximized && renderCardContent()}
```
Comment (jvega190) [inline ui/app/src/components/DashletCard/DashletCard.tsx]: Card content rendered outside of accordion in maximized mode due to fixed height in accordion preventing the content to fully display

File: ui/app/src/components/DashletCard/DashletCard.tsx [hunk 3]
```diff
@@ -23,6 +23,14 @@ import CardContent, { CardContentProps } from '@mui/material/CardContent';
...
 import { UNDEFINED } from '../../utils/constants';
 import CardActions from '@mui/material/CardActions';
+import OpenInFullOutlinedIcon from '@mui/icons-material/OpenInFullOutlined';
+import CloseFullscreenOutlinedIcon from '@mui/icons-material/CloseFullscreenOutlined';
+import IconButton from '@mui/material/IconButton';
+import useSiteDashboardContext from '../SiteDashboard/useSiteDashboardContext';
+import Accordion from '@mui/material/Accordion';
+import AccordionDetails from '@mui/material/AccordionDetails';
+import ExpandMoreIcon from '@mui/icons-material/KeyboardArrowDown';
+import AccordionSummary from '@mui/material/AccordionSummary';
```
Comment (rart) [inline ui/app/src/components/DashletCard/DashletCard.tsx]: Rollback the use of accordion. Just wrap the body area in a conditonal. Can try a [transition component](https://mui.com/material-ui/transitions/) (e.g. Grow) or keep it simple. Also, don't add an icon for the collapsing for _desktop_.

Alternatively, rather use the Accordion alone — no card components. Accordion is a bit bloated in styles and features so think easiest to stick with card for now.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/DashletCard/DashletCard.tsx [hunk 4]
```diff
@@ -55,10 +64,27 @@ export function DashletCard(props: DashletCardProps) {
...
   } = props;
+  // endregion
   const cardHeaderHeight = 62;
   const renderCardHeader = Boolean(title) || Boolean(headerAction);
+  const [isMaximized, setIsMaximized] = useState(false);
+  const dashboardState = useSiteDashboardContext();
+  const theme = useTheme();
+  const desktopScreen = useMediaQuery(theme.breakpoints.up('md'));
+  const [isExpanded, setIsExpanded] = useState(true);
+  const showCardBody = isExpanded || desktopScreen || isMaximized;
```
Comment (rart) [inline ui/app/src/components/DashletCard/DashletCard.tsx]: It's ok to collapse on "desktopScreen". Only not when maximized.

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/components/DashletCard/DashletCard.tsx [hunk 5]
```diff
@@ -69,61 +91,98 @@ export function DashletCard(props: DashletCardProps) {
...
+                <IconButton onClick={() => updateMaximized()}>
+                  {isMaximized ? <CloseFullscreenOutlinedIcon /> : <OpenInFullOutlinedIcon />}
+                </IconButton>
+              )}
+              {headerAction}
+              {!isMaximized && (
+                <IconButton onClick={() => setIsExpanded(!isExpanded)}>
+                  {isExpanded ? <ExpandLessOutlinedIcon /> : <ExpandMoreOutlinedIcon />}
+                </IconButton>
+              )}
```
Comment (rart) [inline ui/app/src/components/DashletCard/DashletCard.tsx]: I'm thinking... let's make the whole header collapse/expand. Just make sure the actual button clicks stop the event so they don't unintentionally collapse the widget.

What if we wrap the CardHeader in CardActionArea.

Let's get rid of the icon/button all together; though not as discoverable, I'm hoping the action area highlight would help with that.

**Review by rart — CHANGES_REQUESTED** [review]

**Review by rart — APPROVED** [review]

**Pull request merged**